pepper-apis/src/main/java/org/broadinstitute/ddp/route/UserRegistrationRoute.java

```diff
@@ -79,19 +83,20 @@ protected int getValidationErrorStatus() {
         return HttpStatus.SC_BAD_REQUEST;
     }
 
-    // todo arz write route filter that checks secret header
     @Override
     public Object handle(Request request, Response response, UserRegistrationPayload payload) throws Exception {
         checkRequiredPayloadProperties(response, payload);
 
         var doLocalRegistration = payload.isLocalRegistration();
         String auth0ClientId = payload.getAuth0ClientId();
         String studyGuid = payload.getStudyGuid();
+        String invitationGuid = payload.getInvitationGuid();
+        final var auth0UserId = new AtomicReference<String>();
```

Comment:
Curious about why this was moved into an `AtomicReference`?

Reply:
Effectively final requirement on L140:

```
InvitationDto invitation = invitationDto.orElseThrow(() -> new DDPException("Could not find invitation " + invitationGuid + " for user " + auth0UserId));
```

Reply:
I think we could have kept it inside the `TransactionWrapper` call and avoid needing to use `AtomicReference`. It's fine either way.